Source/drlg_l2.cpp

```diff
@@ -1298,8 +1300,9 @@ void DefineRoom(int nX1, int nY1, int nX2, int nY2, BOOL ForceHW)
 
 	if (ForceHW == TRUE) {
 		for (i = nX1; i < nX2; i++) {
+			/// BUGFIX: Should loop j between nY1 and nY2 instead of always using nY1.

```

Comment:
This code is basically non-sensical as is. I was almost wondering if it was a decompilation issue.

My proposed change brings it inline with other levels - set peices and central are usually marked as PROTECTED so that other routines don't trash them, e.g. DRLG_L5GChamber. ForceHW is only set when there's a set peice in the room.

I don't think this affects much. Without this flag, I guess here's a small chance of theme rooms and random objects being placed near set pieces. (the set peices themselves are protected in DRLG_L2SetRoom so arent' affected either way).